engine/servermaster/server.go

```diff
@@ -611,14 +652,79 @@ func (s *Server) createGRPCServer() *grpc.Server {
 	return grpcServer
 }
 
-func (s *Server) createHTTPServer() *http.Server {
-	router := gin.New()
-	openapi := NewOpenAPI(s)
-	RegisterRoutes(router, openapi)
+func (s *Server) createHTTPServer() (*http.Server, error) {
+	grpcMux := runtime.NewServeMux(
+		runtime.WithMarshalerOption(runtime.MIMEWildcard, &runtime.JSONPb{
+			MarshalOptions:   protojson.MarshalOptions{UseProtoNames: true},
+			UnmarshalOptions: protojson.UnmarshalOptions{},
+		}),
+	)
+	if err := pb.RegisterJobManagerHandlerServer(context.Background(), grpcMux, s); err != nil {
+		return nil, errors.Trace(err)
+	}
+	if err := pb.RegisterDiscoveryHandlerServer(context.Background(), grpcMux, s); err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	router := http.NewServeMux()
+	RegisterRoutes(router, grpcMux, s.forwardJobAPI)
 
 	return &http.Server{
 		Handler: router,
+	}, nil
+}
+
+func (s *Server) forwardJobAPI(w http.ResponseWriter, r *http.Request) {
+	if err := s.handleForwardJobAPI(w, r); err != nil {
+		se, ok := status.FromError(rpcerror.ToGRPCError(err))
```

Comment:
What does `se` mean?

Reply:
Copy from https://github.com/grpc/grpc-go/blob/3f5b7ab48ca0e597815c6ade7c6c43e8eee95893/status/status.go#L91 I'm not sure what it mean 🤣, just to avoid colliding with `s *Server`.

Reply:
I rename it to `st`. Maybe it's more reasonable.